int32 OnOffControl
float64 reference
float64 GoalmsTime

